### PR TITLE
Fix TFLEDModel

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

The computation of `diagonal_mask` in `TFLEDEncoderSelfAttention`, and `inputs["attention_mask"]` in `TFLEDEncoder` differ from `LEDEncoderSelfAttention` and `LEDEncoder`.

This cause the `logits` and `decoder hidden states` differs in the range `1e-5 ~ 5e-5`, and `encoder hidden states` differs in the range `1e-2 ~ 1e-1`.

This PR fixes these inconsistency, which reduces the difference to the range `1e-7 ~ 1e-6`. I can provide an example if you prefer to see one.

(#15256)

### CODE REVIEW

File: src/transformers/models/longformer/modeling_tf_longformer.py [hunk 1]
```diff
@@ -755,10 +757,15 @@ def call(
             query_vectors, key_vectors, self.one_sided_attn_window_size
         )
 
+        # values to pad for attention probs
+        remove_from_windowed_attention_mask = attention_mask != 0
+        # cast to fp32/fp16 then replace 1's with -inf
+        float_mask = tf.cast(remove_from_windowed_attention_mask, dtype=query_vectors.dtype) * LARGE_NEGATIVE
+
```
Comment (ydshieh) [inline src/transformers/models/longformer/modeling_tf_longformer.py:764]: This is not logically necessary, since `TFLongformerMainLayer` prepares `attention_mask` slightly different from `LongformerModel`

PT
https://github.com/huggingface/transformers/blob/d7136abe59f2183ab4efb343ae249b089be83681/src/transformers/models/longformer/modeling_longformer.py#L1688

TF (note the `abs`)
https://github.com/huggingface/transformers/blob/d7136abe59f2183ab4efb343ae249b089be83681/src/transformers/models/longformer/modeling_tf_longformer.py#L1745

Need to keep this for `# copied from` in `TFLEDEncoderSelfAttention`, because for that model, this block is necessary - the reason is: `tf.abs` is not used when preparing the attention mask there

https://github.com/huggingface/transformers/blob/d7136abe59f2183ab4efb343ae249b089be83681/src/transformers/models/led/modeling_tf_led.py#L103

**Comment by LysandreJik** [comment]
@patrickvonplaten was very involved in those two models, but @Rocketknight1 and @gante may also take a look

**Review by gante — APPROVED** [review]
It looks good to me, as it recovers a pattern lost in the transition and lowers the error. I'm assuming this PR is also a dependency of https://github.com/huggingface/transformers/pull/15256, correct?

Waiting for a comment from @patrickvonplaten before merging

**Comment by ydshieh** [comment]
> It looks good to me, as it recovers a pattern lost in the transition and lowers the error. I'm assuming this PR is also a dependency of #15256, correct?
> 

Yes!

**Comment by patrickvonplaten** [comment]
Looks clean - thanks a mille for diving into this complicated code @ydshieh! Could we maybe add a test between PT and TF for LED that would have failed previously without this change? Also Longformer should be affected by this problem as well no?

**Comment by ydshieh** [comment]
> Looks clean - thanks a mille for diving into this complicated code @ydshieh! Could we maybe add a test between PT and TF for LED that would have failed previously without this change? Also Longformer should be affected by this problem as well no?

For Longformer, 

 - about `merge `global_attention_mask` and `attention_mask``: TF version works correctly in `_merge_to_attention_mask`.
 - about the `diagonal_mask`, it turns out `TFLongformerMainLayer` use something different `extended_attention_mask = tf.cast(tf.math.abs(1 - extended_attention_mask), tf.dtypes.float32) * -10000.0` (note the `tf.math.abs` here), which makes all of masked places negative (including global attn), while in `TFLED`, it uses `_expand_mask`, which make global attn `positive`.
This difference makes `TFLongformer` gives the same results as PT models in the test. (I feel there is something strange here though!)

File: tests/test_modeling_tf_led.py [hunk 2]
```diff
@@ -362,6 +363,128 @@ def check_encoder_attentions_output(outputs):
             self.assertEqual(model.config.output_hidden_states, True)
             check_encoder_attentions_output(outputs)
 
+    # TODO: Remove this once a more thorough pt/tf equivalence could be implemented in `test_modeling_tf_common.py`.
+    # (Currently, such a test will fail some other model tests: it requires some time to fix them.)
+    @is_pt_tf_cross_test
+    def test_pt_tf_model_equivalence_extra(self):
+        import torch
```
Comment (ydshieh) [inline tests/test_modeling_tf_led.py:370]: @patrickvonplaten  I added this test for TFLED. Is this OK?

Comment (patrickvonplaten) [reply tests/test_modeling_tf_led.py:370]: Perfect!

**Review by patrickvonplaten — APPROVED** [review]
